### PR TITLE
Measure Operation time using `console` object

### PR DESCRIPTION
**Pull request by SarjuHansaliya**
## TL;DR;

Measure Operation time using `console` object
## Username

[SarjuHansaliya](https://github.com/SarjuHansaliya)

### CODE REVIEW

File: _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+tip-number: xx
+tip-username: SarjuHansaliya
+tip-username-profile: https://github.com/SarjuHansaliya
+tip-tldr: Measure Operation time using `console` object
+
+categories:
+    - en
+---
+
+Sometime we need to measure time of some tasks to analyze how much time it is taking to execute.To achieve that normal practice is to notedown time before starting of the task and notedown time after completion of the task , and then take a difference between two times.But that is lots of work for only time measuring.
```
Comment (kurtextrem) [inline _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md]: "how much time it takes to execute"
Space after dot.
"To achieve that, "
"note down"
No space before comma.
"this is a lot of work"

File: _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Simple method is to do this thing is to use `console.time(label)` and `console.timeEnd(label)`.Here is the snippet.
```
Comment (kurtextrem) [inline _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md]: "The most simplest way to achieve this is" would be better.
Space after dot.

File: _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+```javascript
+// `timer` is a label which will be used to end particular timer.
```
Comment (kurtextrem) [inline _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md]: "to end that particular timer"

File: _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+// Label can be any valid string.
+console.time("timer"); 
+YOUR_TASK();
+console.timeEnd("timer");
+
+//Output:: timer: x ms 
+```
+> Note : As [Mozilla](https://developer.mozilla.org/en-US/docs/Web/API/Console/time) suggested that don't use this for production sites , only use it for development purpose only.
```
Comment (kurtextrem) [inline _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md]: No space before comma.

**Comment by SarjuHansaliya** [comment]
@kurtextrem Thanks.
Is it good now?

File: _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+tip-number: xx
+tip-username: SarjuHansaliya
+tip-username-profile: https://github.com/SarjuHansaliya
+tip-tldr: Measure Operation time using `console` object
+
+categories:
+    - en
+---
+
+Sometime we need to measure time of some tasks to analyze how much time it takes to execute. To achieve that, normal practice is to note down time before starting of the task and note down time after completion of the task, and then take a difference between two times.But this is a lot of work for only time measuring.
```
Comment (kurtextrem) [inline _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md]: "starting the task" and "completing the task" is better.
"take the difference" 
Space after dot missing. (Before "But")

File: _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+The most simplest way to achieve this is to use `console.time(label)` and `console.timeEnd(label)`.Here is the snippet.
```
Comment (kurtextrem) [inline _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md]: Space after dot is missing (before "Here")

File: _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+```javascript
+// `timer` is a label which will be used to end that particular timer.
+// Label can be any valid string.
+console.time("timer"); 
+YOUR_TASK();
+console.timeEnd("timer");
+
+//Output:: timer: x ms 
+```
+> Note : As [Mozilla](https://developer.mozilla.org/en-US/docs/Web/API/Console/time) suggested that don't use this for production sites, only use it for development purpose only.
```
Comment (kurtextrem) [inline _posts/en/2016-xx-xx-measure-operation-time-using-console-object.md]: No space before `:`, no "that" after "suggested"
", use it for development purposes only"

**Comment by SarjuHansaliya** [comment]
@kurtextrem  thanks :+1:

**Comment by loverajoel** [comment]
@SarjuHansaliya There is one tip equal than yours, take a look [here](http://www.jstips.co/en/tip-to-measure-performance-of-a-javascript-block/).
What about merge both?

**Comment by SarjuHansaliya** [comment]
Ohh
Sure , We should merge both.
Sorry I haven't seen that post earlier

**Comment by zenopopovici** [comment]
@SarjuHansaliya Just make changes on that one and submit a new PR.

**Comment by SarjuHansaliya** [comment]
Can you help me ?
I don't see any "Create Pull Request" Button.
screen shot 2016-03-12 at 10 06 35 am

Do I have to close this PR ?

**Comment by gromgit** [comment]
@SarjuHansaliya I suggest you leave this PR alone for now, just start a new one on the other tip. Then when you're done submitting that PR, come back here and close this one.

**Comment by SarjuHansaliya** [comment]
@gromgit 
I am new to git , so I am little bit confused.
I have made changes to tip 13 in my local , then I push it to my forked repo on github.
Now those commits are listed here , why so ? 

Even I can't see new PR button ?

Can you guide me?

**Comment by gromgit** [comment]
@SarjuHansaliya Ah, I didn't notice the problem earlier. GitHub tracks PR content by **branch**; since all your changes were on the main `gh-pages` branch, what you ended up doing was combine all your changes (even for other tips) into this PR instead.

The cleanest and most reliable way to resolve this is to close this PR and reset your own state:
1. Copy out any changed files from your old fork to another directory.
2. [Delete your old fork on GitHub](https://help.github.com/articles/deleting-a-repository/).
3. Delete the local copy of your own fork.
4. Re-fork JSTips.
5. Copy in all the changed files you saved from [1].

Going forward, you really should create a separate branch for each PR. Here's what I do:
1. [Update local fork from the main repo](https://help.github.com/articles/syncing-a-fork/).
2. Create and checkout a new branch: `git checkout -b tipXX-update gh-pages`
3. Make and commit edits.
4. Push this branch to your fork: `git push upstream tipXX-update`
5. Go to your forked repo page. There should be a Compare and Pull Request button next to your new branch. Click it and follow the instructions.

**Comment by SarjuHansaliya** [comment]
@gromgit Oh thanks man.
Now I got the problem , It makes me clear now.

So I am closing this PR.
Will submit new.

**Pull request closed without merging**